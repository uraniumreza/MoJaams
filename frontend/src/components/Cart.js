import React from 'react';
import PropTypes from 'prop-types';

const Cart = ({
  cart,
  editOrderItem,
  removeFromCart,
  goBack,
  goNext,
  step,
}) => {
  return (
    <div className={`full-page ${step === 3 ? 'visible' : ''}`}>
      {step === 3 && (
        <>
          <h4 className="title">Your Order!</h4>
          <div className="order-items-container">
            {Object.entries(cart).map(([id, orderItem]) => (
              <div className="order-item" key={id}>
                <span className="name">
                  {`${orderItem.itemName} - ${orderItem.itemVariant} `}
                  <strong>(x{orderItem.quantity})</strong>
                </span>
                <div>
                  <button key="update" onClick={() => editOrderItem(id)}>
                    UPDATE
                  </button>
                  <button key="delete" onClick={() => removeFromCart(id)}>
                    DELETE
                  </button>
                </div>
              </div>
            ))}
            {!Object.entries(cart).length && (
              <div className="order-item empty">
                <div className="name empty">Nothing's Here! ¯\_(ツ)_/¯</div>
                <button onClick={goBack}>ADD PIZZA</button>
              </div>
            )}
          </div>

          {Object.entries(cart).length && (
            <div className="navigation-container">
              <button className="nav-btn back" onClick={goBack}>
                ADD MORE
              </button>
              <button className="nav-btn next" onClick={goNext}>
                CONFIRM
              </button>
            </div>
          )}
        </>
      )}
    </div>
  );
};

Cart.propTypes = {
  step: PropTypes.number.isRequired,
  cart: PropTypes.objectOf(
    PropTypes.shape({
      itemName: PropTypes.string.isRequired,
      itemVariant: PropTypes.string.isRequired,
      quantity: PropTypes.number.isRequired,
    }),
  ),
  editOrderItem: PropTypes.func.isRequired,
  removeFromCart: PropTypes.func.isRequired,
  goBack: PropTypes.func.isRequired,
  goNext: PropTypes.func.isRequired,
};

export default Cart;
