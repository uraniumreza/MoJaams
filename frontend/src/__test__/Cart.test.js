import React from 'react';
import renderer from 'react-test-renderer';

import Cart from '../components/Cart';

describe('Cart - SnapShot test', () => {
  it('renders correctly when there are no items', () => {
    const tree = renderer
      .create(
        <Cart
          step={3}
          cart={{}}
          editOrderItem={() => {}}
          removeFromCart={() => {}}
          goBack={() => {}}
          goNext={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when there are 2 items', () => {
    const tree = renderer
      .create(
        <Cart
          step={3}
          cart={{
            '3': {
              itemName: 'Margarita',
              itemVariant: 'Small',
              quantity: 1,
            },
            '5': {
              itemName: 'Four Seasons',
              itemVariant: 'Medium',
              quantity: 2,
            },
          }}
          editOrderItem={() => {}}
          removeFromCart={() => {}}
          goBack={() => {}}
          goNext={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
