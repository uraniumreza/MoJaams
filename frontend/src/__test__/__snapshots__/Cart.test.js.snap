// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart - SnapShot test renders correctly when there are 2 items 1`] = `
<div
  className="full-page visible"
>
  <h4
    className="title"
  >
    Your Order!
  </h4>
  <div
    className="order-items-container"
  >
    <div
      className="order-item"
    >
      <span
        className="name"
      >
        Margarita - Small 
        <strong>
          (x
          1
          )
        </strong>
      </span>
      <div>
        <button
          onClick={[Function]}
        >
          UPDATE
        </button>
        <button
          onClick={[Function]}
        >
          DELETE
        </button>
      </div>
    </div>
    <div
      className="order-item"
    >
      <span
        className="name"
      >
        Four Seasons - Medium 
        <strong>
          (x
          2
          )
        </strong>
      </span>
      <div>
        <button
          onClick={[Function]}
        >
          UPDATE
        </button>
        <button
          onClick={[Function]}
        >
          DELETE
        </button>
      </div>
    </div>
  </div>
  <div
    className="navigation-container"
  >
    <button
      className="nav-btn back"
      onClick={[Function]}
    >
      ADD MORE
    </button>
    <button
      className="nav-btn next"
      onClick={[Function]}
    >
      CONFIRM
    </button>
  </div>
</div>
`;

exports[`Cart - SnapShot test renders correctly when there are no items 1`] = `
<div
  className="full-page visible"
>
  <h4
    className="title"
  >
    Your Order!
  </h4>
  <div
    className="order-items-container"
  >
    <div
      className="order-item"
    >
      <div
        className="name"
      >
        Nothing's Here! ¯\\_(ツ)_/¯
      </div>
      <button
        onClick={[Function]}
      >
        ADD PIZZA
      </button>
    </div>
  </div>
  0
</div>
`;
